deeplearning4j-nn/src/main/java/org/deeplearning4j/datasets/iterator/parallel/MultiBoolean.java

```diff
@@ -0,0 +1,93 @@
+package org.deeplearning4j.datasets.iterator.parallel;
+
+import lombok.extern.slf4j.Slf4j;
+import org.nd4j.linalg.exception.ND4JIllegalStateException;
+
+/**
+ * This is utility class, that allows easy handling of multiple joint boolean states
+ *
+ * @author raver119@gmail.com
+ */
+@Slf4j
+public class MultiBoolean {
```

Comment:
Non-thread safe/atomic here? And what's the maximum - 32 values? (This should be in the javadoc)
It's a little inefficient in terms of space, but extending AtomicIntegerArray would be an alternative.